go/proof.go

```diff
@@ -309,6 +309,42 @@ func getPadding(spec *InnerSpec, branch int32) (minPrefix, maxPrefix, suffix int
 	return
 }
 
+// leftBranchesAreEmpty returns true if the padding bytes correspond to all empty children
+// on the left side of this branch, ie. it's a valid placeholder on a leftmost path
+func leftBranchesAreEmpty(spec *InnerSpec, op *InnerOp, branch int32) bool {
+	idx := getPosition(spec.ChildOrder, branch)
+	// compare the prefix bytes with the appropriate number of empty children
+	leftChildren := len(spec.ChildOrder) - 1 - idx
```

Comment:
Looking at the failing test cases, I think this math is wrong.

Assume there are 2 children, those would have position 0, and 1.

For position 0, I would get 2-1-0 = 1 left children (but should be 0)
For position 1, I would get 2-1-1 = 0 left children. (but should be 1)

I think `leftChildren := idx` would be correct

Reply:
Yeah, this makes more sense. I was thinking about this backwards, such that `branch` meant, "the index over here" rather than, "the index corresponding to sibling data with this shape"